lib/processor/zephyr_native/atomic.h

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * @file	zephyr_native/atomic.h
+ * @brief	Zephyr native test environment atomic primitives for libmetal.
+ */
+
+#ifndef __METAL_ZEPHYR_NATIVE_ATOMIC__H__
+#define __METAL_ZEPHYR_NATIVE_ATOMIC__H__
+
+#endif /* __METAL_ZEPHYR_NATIVE_ATOMIC__H__ */
```

Comment:
I would really like to replace all these do nothing headers with a single copy of a generic version.  For "cpus" where both files are do nothing they can just define cpu_generic or something like that.

For this case you have something real to do in cpu.h so the cmake can select the generic-atomic.h and you can have the real cpu.h below.

Reply:
I can do that in a separate commit if you want